### DISCUSSION TITLE
How can I list all short URLs?

### DISCUSSION THREAD

**Discussion by rosano** [YOURLS/YOURLS#3033] [Q&A] [answered]
I would like to display all slugs for consumption by a 3rd party service. Would this be done via the API or a plugin or is it built in somewhere?

**Comment by ozh** [comment]
This isn't built in.

You would need to query the table `yourls_url` and retrieve the keywords, ie `SELECT keyword FROM yourls_url`. See https://github.com/YOURLS/YOURLS/wiki/How-to-write-SQL-queries-in-YOURLS

Doing it directly or within YOURLS using the API depends on your use case. You can easily make plugins to add custom actions to the API, see https://github.com/YOURLS/API-action

**Comment by rosano** [comment, marked as answer]
Thanks. I made a custom API action via plugin https://github.com/rosano/API-action/blob/master/plugin.php

**Reply by ozh** [reply]
Nice. For better portability you should not hardcode the `yourls_url` table name, but use user defined constant YOURLS_DB_TABLE_URL

Once you clean up the readme to explain how your plugin works, feel free to send a pull request against https://github.com/YOURLS/awesome-yourls to get your plugin listed !

**Reply by rosano** [reply]
Done :)
